pytorch_lightning/metrics/metric.py

```diff
@@ -302,6 +297,199 @@ def state_dict(self, *args, **kwargs):
                 state_dict.update({key: current_val})
         return state_dict
 
+    def _filter_kwargs(self, **kwargs):
+        """ filter kwargs such that they match the update signature of the metric """
+
+        # filter all parameters based on update signature except those of
+        # type VAR_POSITIONAL (*args) and VAR_KEYWORD (**kwargs)
+        filtered_kwargs = {
+            k: v
+            for k, v in kwargs.items()
+            if k in self._update_signature.parameters.keys()
+            and self._update_signature.parameters[k].kind not in (inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD)
+        }
+
+        # if no kwargs filtered, return al kwargs as default
+        if not filtered_kwargs:
+            filtered_kwargs = kwargs
+        return filtered_kwargs
+
+    def __hash__(self):
+        hash_vals = [self.__class__.__name__]
+
+        for key in self._defaults.keys():
+            hash_vals.append(getattr(self, key))
+
+        return hash(tuple(hash_vals))
+
+    def __add__(self, other: Any):
+        from pytorch_lightning.metrics.compositional import CompositionalMetric
```

Comment:
why do we need to import it here? bc cyclic import?

Reply:
exactly